src/stream/futures.rs

```diff
@@ -0,0 +1,67 @@
+use future::{Future, IntoFuture};
+use stream::Stream;
+use poll::Poll;
+use Async;
+
+use std::prelude::v1::*;
+
+/// An adaptor for a stream of futures to execute the futures concurrently, if
+/// possible, delivering results as they become available.
+///
+/// This adaptor will return their results in the order that they complete.
+/// This is created by the `futures` method.
+///
+#[must_use = "streams do nothing unless polled"]
+pub struct Futures<F>
+    where F: Future
+{
+    futures: Vec<F>,
+}
+
+/// Converts a `Vec` of `IntoFuture` elements into a `Stream` over the underlying
+/// futures. The created stream will yield into values produced by those futures.
+/// Similar to `buffer_unordered` the order of the produced items will not be equal
+/// with the order of elements in the input vector.
+///
+pub fn futures<I, It, F>(futures: I) -> Futures<F::Future>
+    where I: IntoIterator<Item = F, IntoIter = It>,
+          It: Iterator<Item = F> + ExactSizeIterator,
+          F: IntoFuture
+{
+    Futures {
+        futures: futures.into_iter().map(IntoFuture::into_future).collect(),
```

Comment:
Will the compiler really optimize away this allocation?  Seems simpler to me to just have the caller pass in a `Vec`, it's a much more straightforward signature as well.